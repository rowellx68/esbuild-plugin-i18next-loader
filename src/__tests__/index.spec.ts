import { test, expect } from 'vitest'
import path from 'node:path'
import i18nexPlugin from '../index'
import esbuild from 'esbuild'

test('no module resolution', async () => {
  const result = await esbuild.build({
    entryPoints: [path.resolve(__dirname, '__fixtures__/component.ts')],
    bundle: true,
    write: false,
    plugins: [
      i18nexPlugin({
        paths: [path.resolve(__dirname, '__fixtures__/locales')],
      }),
    ],
  })

  expect(result).toMatchSnapshot()
})

test('basename', async () => {
  const result = await esbuild.build({
    entryPoints: [path.resolve(__dirname, '__fixtures__/component.ts')],
    bundle: true,
    write: false,
    plugins: [
      i18nexPlugin({
        namespaceResolution: 'basename',
        paths: [path.resolve(__dirname, '__fixtures__/locales')],
      }),
    ],
  })

  expect(result).toMatchSnapshot()
})

test('relativePath', async () => {
  const result = await esbuild.build({
    entryPoints: [path.resolve(__dirname, '__fixtures__/component.ts')],
    bundle: true,
    write: false,
    plugins: [
      i18nexPlugin({
        namespaceResolution: 'relativePath',
        paths: [path.resolve(__dirname, '__fixtures__/locales')],
      }),
    ],
  })

  expect(result).toMatchSnapshot()
})