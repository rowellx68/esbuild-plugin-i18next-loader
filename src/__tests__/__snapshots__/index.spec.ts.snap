// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`basename 1`] = `
{
  "errors": [],
  "mangleCache": undefined,
  "metafile": undefined,
  "outputFiles": [
    {
      "contents": Uint8Array [
        34,
        117,
        115,
        101,
        32,
        115,
        116,
        114,
        105,
        99,
        116,
        34,
        59,
        10,
        40,
        40,
        41,
        32,
        61,
        62,
        32,
        123,
        10,
        32,
        32,
        47,
        47,
        32,
        105,
        49,
        56,
        110,
        101,
        120,
        116,
        58,
        108,
        111,
        99,
        97,
        108,
        105,
        122,
        97,
        116,
        105,
        111,
        110,
        115,
        10,
        32,
        32,
        118,
        97,
        114,
        32,
        108,
        111,
        99,
        97,
        108,
        105,
        122,
        97,
        116,
        105,
        111,
        110,
        115,
        95,
        100,
        101,
        102,
        97,
        117,
        108,
        116,
        32,
        61,
        32,
        123,
        32,
        34,
        101,
        110,
        45,
        71,
        66,
        34,
        58,
        32,
        123,
        32,
        34,
        116,
        114,
        97,
        110,
        115,
        108,
        97,
        116,
        105,
        111,
        110,
        115,
        34,
        58,
        32,
        123,
        32,
        34,
        109,
        101,
        115,
        115,
        97,
        103,
        101,
        34,
        58,
        32,
        34,
        72,
        101,
        108,
        108,
        111,
        32,
        87,
        111,
        114,
        108,
        100,
        33,
        34,
        32,
        125,
        44,
        32,
        34,
        111,
        116,
        104,
        101,
        114,
        45,
        110,
        97,
        109,
        101,
        115,
        112,
        97,
        99,
        101,
        34,
        58,
        32,
        123,
        32,
        34,
        111,
        116,
        104,
        101,
        114,
        34,
        58,
        32,
        123,
        32,
        34,
        109,
        101,
        115,
        115,
        97,
        103,
        101,
        34,
        58,
        32,
        34,
        72,
        105,
        32,
        116,
        104,
        101,
        114,
        101,
        33,
        34,
        32,
        125,
        32,
        125,
        32,
        125,
        44,
        32,
        34,
        101,
        115,
        45,
        69,
        83,
        34,
        58,
        32,
        123,
        32,
        34,
        116,
        114,
        97,
        110,
        115,
        108,
        97,
        116,
        105,
        111,
        110,
        115,
        34,
        58,
        32,
        123,
        32,
        34,
        109,
        101,
        115,
        115,
        97,
        103,
        101,
        34,
        58,
        32,
        34,
        92,
        120,
        65,
        49,
        72,
        111,
        108,
        97,
        32,
        77,
        117,
        110,
        100,
        111,
        33,
        34,
        32,
        125,
        44,
        32,
        34,
        111,
        116,
        104,
        101,
        114,
        45,
        110,
        97,
        109,
        101,
        115,
        112,
        97,
        99,
        101,
        34,
        58,
        32,
        123,
        32,
        34,
        111,
        116,
        104,
        101,
        114,
        34,
        58,
        32,
        123,
        32,
        34,
        109,
        101,
        115,
        115,
        97,
        103,
        101,
        34,
        58,
        32,
        34,
        92,
        120,
        65,
        49,
        72,
        111,
        108,
        97,
        33,
        34,
        32,
        125,
        32,
        125,
        32,
        125,
        32,
        125,
        59,
        10,
        10,
        32,
        32,
        47,
        47,
        32,
        115,
        114,
        99,
        47,
        95,
        95,
        116,
        101,
        115,
        116,
        115,
        95,
        95,
        47,
        95,
        95,
        102,
        105,
        120,
        116,
        117,
        114,
        101,
        115,
        95,
        95,
        47,
        99,
        111,
        109,
        112,
        111,
        110,
        101,
        110,
        116,
        46,
        116,
        115,
        10,
        32,
        32,
        118,
        97,
        114,
        32,
        99,
        111,
        109,
        112,
        111,
        110,
        101,
        110,
        116,
        95,
        100,
        101,
        102,
        97,
        117,
        108,
        116,
        32,
        61,
        32,
        40,
        41,
        32,
        61,
        62,
        32,
        123,
        10,
        32,
        32,
        32,
        32,
        114,
        101,
        116,
        117,
        114,
        110,
        32,
        108,
        111,
        99,
        97,
        108,
        105,
        122,
        97,
        116,
        105,
        111,
        110,
        115,
        95,
        100,
        101,
        102,
        97,
        117,
        108,
        116,
        59,
        10,
        32,
        32,
        125,
        59,
        10,
        125,
        41,
        40,
        41,
        59,
        10,
      ],
      "hash": "QfV9OimLkM8",
      "path": "<stdout>",
      "text": ""use strict";
(() => {
  // i18next:localizations
  var localizations_default = { "en-GB": { "translations": { "message": "Hello World!" }, "other-namespace": { "other": { "message": "Hi there!" } } }, "es-ES": { "translations": { "message": "\\xA1Hola Mundo!" }, "other-namespace": { "other": { "message": "\\xA1Hola!" } } } };

  // src/__tests__/__fixtures__/component.ts
  var component_default = () => {
    return localizations_default;
  };
})();
",
    },
  ],
  "warnings": [],
}
`;

exports[`no module resolution 1`] = `
{
  "errors": [],
  "mangleCache": undefined,
  "metafile": undefined,
  "outputFiles": [
    {
      "contents": Uint8Array [
        34,
        117,
        115,
        101,
        32,
        115,
        116,
        114,
        105,
        99,
        116,
        34,
        59,
        10,
        40,
        40,
        41,
        32,
        61,
        62,
        32,
        123,
        10,
        32,
        32,
        47,
        47,
        32,
        105,
        49,
        56,
        110,
        101,
        120,
        116,
        58,
        108,
        111,
        99,
        97,
        108,
        105,
        122,
        97,
        116,
        105,
        111,
        110,
        115,
        10,
        32,
        32,
        118,
        97,
        114,
        32,
        108,
        111,
        99,
        97,
        108,
        105,
        122,
        97,
        116,
        105,
        111,
        110,
        115,
        95,
        100,
        101,
        102,
        97,
        117,
        108,
        116,
        32,
        61,
        32,
        123,
        32,
        34,
        101,
        110,
        45,
        71,
        66,
        34,
        58,
        32,
        123,
        32,
        34,
        109,
        101,
        115,
        115,
        97,
        103,
        101,
        34,
        58,
        32,
        34,
        72,
        101,
        108,
        108,
        111,
        32,
        87,
        111,
        114,
        108,
        100,
        33,
        34,
        44,
        32,
        34,
        111,
        116,
        104,
        101,
        114,
        34,
        58,
        32,
        123,
        32,
        34,
        109,
        101,
        115,
        115,
        97,
        103,
        101,
        34,
        58,
        32,
        34,
        72,
        105,
        32,
        116,
        104,
        101,
        114,
        101,
        33,
        34,
        32,
        125,
        32,
        125,
        44,
        32,
        34,
        101,
        115,
        45,
        69,
        83,
        34,
        58,
        32,
        123,
        32,
        34,
        109,
        101,
        115,
        115,
        97,
        103,
        101,
        34,
        58,
        32,
        34,
        92,
        120,
        65,
        49,
        72,
        111,
        108,
        97,
        32,
        77,
        117,
        110,
        100,
        111,
        33,
        34,
        44,
        32,
        34,
        111,
        116,
        104,
        101,
        114,
        34,
        58,
        32,
        123,
        32,
        34,
        109,
        101,
        115,
        115,
        97,
        103,
        101,
        34,
        58,
        32,
        34,
        92,
        120,
        65,
        49,
        72,
        111,
        108,
        97,
        33,
        34,
        32,
        125,
        32,
        125,
        32,
        125,
        59,
        10,
        10,
        32,
        32,
        47,
        47,
        32,
        115,
        114,
        99,
        47,
        95,
        95,
        116,
        101,
        115,
        116,
        115,
        95,
        95,
        47,
        95,
        95,
        102,
        105,
        120,
        116,
        117,
        114,
        101,
        115,
        95,
        95,
        47,
        99,
        111,
        109,
        112,
        111,
        110,
        101,
        110,
        116,
        46,
        116,
        115,
        10,
        32,
        32,
        118,
        97,
        114,
        32,
        99,
        111,
        109,
        112,
        111,
        110,
        101,
        110,
        116,
        95,
        100,
        101,
        102,
        97,
        117,
        108,
        116,
        32,
        61,
        32,
        40,
        41,
        32,
        61,
        62,
        32,
        123,
        10,
        32,
        32,
        32,
        32,
        114,
        101,
        116,
        117,
        114,
        110,
        32,
        108,
        111,
        99,
        97,
        108,
        105,
        122,
        97,
        116,
        105,
        111,
        110,
        115,
        95,
        100,
        101,
        102,
        97,
        117,
        108,
        116,
        59,
        10,
        32,
        32,
        125,
        59,
        10,
        125,
        41,
        40,
        41,
        59,
        10,
      ],
      "hash": "IjM5lih85PY",
      "path": "<stdout>",
      "text": ""use strict";
(() => {
  // i18next:localizations
  var localizations_default = { "en-GB": { "message": "Hello World!", "other": { "message": "Hi there!" } }, "es-ES": { "message": "\\xA1Hola Mundo!", "other": { "message": "\\xA1Hola!" } } };

  // src/__tests__/__fixtures__/component.ts
  var component_default = () => {
    return localizations_default;
  };
})();
",
    },
  ],
  "warnings": [],
}
`;

exports[`relativePath 1`] = `
{
  "errors": [],
  "mangleCache": undefined,
  "metafile": undefined,
  "outputFiles": [
    {
      "contents": Uint8Array [
        34,
        117,
        115,
        101,
        32,
        115,
        116,
        114,
        105,
        99,
        116,
        34,
        59,
        10,
        40,
        40,
        41,
        32,
        61,
        62,
        32,
        123,
        10,
        32,
        32,
        47,
        47,
        32,
        105,
        49,
        56,
        110,
        101,
        120,
        116,
        58,
        108,
        111,
        99,
        97,
        108,
        105,
        122,
        97,
        116,
        105,
        111,
        110,
        115,
        10,
        32,
        32,
        118,
        97,
        114,
        32,
        108,
        111,
        99,
        97,
        108,
        105,
        122,
        97,
        116,
        105,
        111,
        110,
        115,
        95,
        100,
        101,
        102,
        97,
        117,
        108,
        116,
        32,
        61,
        32,
        123,
        32,
        34,
        101,
        110,
        45,
        71,
        66,
        34,
        58,
        32,
        123,
        32,
        34,
        116,
        114,
        97,
        110,
        115,
        108,
        97,
        116,
        105,
        111,
        110,
        115,
        34,
        58,
        32,
        123,
        32,
        34,
        109,
        101,
        115,
        115,
        97,
        103,
        101,
        34,
        58,
        32,
        34,
        72,
        101,
        108,
        108,
        111,
        32,
        87,
        111,
        114,
        108,
        100,
        33,
        34,
        32,
        125,
        44,
        32,
        34,
        111,
        116,
        104,
        101,
        114,
        45,
        110,
        97,
        109,
        101,
        115,
        112,
        97,
        99,
        101,
        34,
        58,
        32,
        123,
        32,
        34,
        111,
        116,
        104,
        101,
        114,
        34,
        58,
        32,
        123,
        32,
        34,
        109,
        101,
        115,
        115,
        97,
        103,
        101,
        34,
        58,
        32,
        34,
        72,
        105,
        32,
        116,
        104,
        101,
        114,
        101,
        33,
        34,
        32,
        125,
        32,
        125,
        44,
        32,
        34,
        110,
        101,
        115,
        116,
        101,
        100,
        34,
        58,
        32,
        123,
        32,
        34,
        116,
        114,
        97,
        110,
        115,
        108,
        97,
        116,
        105,
        111,
        110,
        115,
        34,
        58,
        32,
        123,
        32,
        34,
        109,
        101,
        115,
        115,
        97,
        103,
        101,
        34,
        58,
        32,
        34,
        72,
        101,
        108,
        108,
        111,
        32,
        87,
        111,
        114,
        108,
        100,
        33,
        34,
        32,
        125,
        32,
        125,
        32,
        125,
        44,
        32,
        34,
        101,
        115,
        45,
        69,
        83,
        34,
        58,
        32,
        123,
        32,
        34,
        116,
        114,
        97,
        110,
        115,
        108,
        97,
        116,
        105,
        111,
        110,
        115,
        34,
        58,
        32,
        123,
        32,
        34,
        109,
        101,
        115,
        115,
        97,
        103,
        101,
        34,
        58,
        32,
        34,
        92,
        120,
        65,
        49,
        72,
        111,
        108,
        97,
        32,
        77,
        117,
        110,
        100,
        111,
        33,
        34,
        32,
        125,
        44,
        32,
        34,
        111,
        116,
        104,
        101,
        114,
        45,
        110,
        97,
        109,
        101,
        115,
        112,
        97,
        99,
        101,
        34,
        58,
        32,
        123,
        32,
        34,
        111,
        116,
        104,
        101,
        114,
        34,
        58,
        32,
        123,
        32,
        34,
        109,
        101,
        115,
        115,
        97,
        103,
        101,
        34,
        58,
        32,
        34,
        92,
        120,
        65,
        49,
        72,
        111,
        108,
        97,
        33,
        34,
        32,
        125,
        32,
        125,
        44,
        32,
        34,
        110,
        101,
        115,
        116,
        101,
        100,
        34,
        58,
        32,
        123,
        32,
        34,
        116,
        114,
        97,
        110,
        115,
        108,
        97,
        116,
        105,
        111,
        110,
        115,
        34,
        58,
        32,
        123,
        32,
        34,
        109,
        101,
        115,
        115,
        97,
        103,
        101,
        34,
        58,
        32,
        34,
        92,
        120,
        65,
        49,
        72,
        111,
        108,
        97,
        32,
        77,
        117,
        110,
        100,
        111,
        33,
        34,
        32,
        125,
        32,
        125,
        32,
        125,
        32,
        125,
        59,
        10,
        10,
        32,
        32,
        47,
        47,
        32,
        115,
        114,
        99,
        47,
        95,
        95,
        116,
        101,
        115,
        116,
        115,
        95,
        95,
        47,
        95,
        95,
        102,
        105,
        120,
        116,
        117,
        114,
        101,
        115,
        95,
        95,
        47,
        99,
        111,
        109,
        112,
        111,
        110,
        101,
        110,
        116,
        46,
        116,
        115,
        10,
        32,
        32,
        118,
        97,
        114,
        32,
        99,
        111,
        109,
        112,
        111,
        110,
        101,
        110,
        116,
        95,
        100,
        101,
        102,
        97,
        117,
        108,
        116,
        32,
        61,
        32,
        40,
        41,
        32,
        61,
        62,
        32,
        123,
        10,
        32,
        32,
        32,
        32,
        114,
        101,
        116,
        117,
        114,
        110,
        32,
        108,
        111,
        99,
        97,
        108,
        105,
        122,
        97,
        116,
        105,
        111,
        110,
        115,
        95,
        100,
        101,
        102,
        97,
        117,
        108,
        116,
        59,
        10,
        32,
        32,
        125,
        59,
        10,
        125,
        41,
        40,
        41,
        59,
        10,
      ],
      "hash": "J1LUJIIDQN4",
      "path": "<stdout>",
      "text": ""use strict";
(() => {
  // i18next:localizations
  var localizations_default = { "en-GB": { "translations": { "message": "Hello World!" }, "other-namespace": { "other": { "message": "Hi there!" } }, "nested": { "translations": { "message": "Hello World!" } } }, "es-ES": { "translations": { "message": "\\xA1Hola Mundo!" }, "other-namespace": { "other": { "message": "\\xA1Hola!" } }, "nested": { "translations": { "message": "\\xA1Hola Mundo!" } } } };

  // src/__tests__/__fixtures__/component.ts
  var component_default = () => {
    return localizations_default;
  };
})();
",
    },
  ],
  "warnings": [],
}
`;
